### PR TITLE
Add hierarchical BERT model

### PR DESCRIPTION
**Pull request by achyudh**
This is a hierarchical model on top of BERT. This would directly be applicable to relevance transfer, since I am assuming that the primary reason for poor results is the truncation of documents to 512 tokens.

My commit history isn't clean so feel free to squash the changes.

### CODE REVIEW

File: common/trainers/relevance_transfer_trainer.py [hunk 1]
```diff
@@ -1,133 +1,154 @@
...
+                        batch_lengths = batch.text[1][indices]
+                        batch_label = batch.label
+                    else:
+                        batch_text = batch.text[0]
+                        batch_lengths = batch.text[1]
+                        batch_label = batch.label
+
+                if hasattr(self.model, 'tar') and self.model.tar:
+                    if 'ignore_lengths' in self.config and self.config['ignore_lengths']:
+                        logits, rnn_outs = torch.sigmoid(self.model(batch_text)).squeeze(dim=1)
```
Comment (daemon) [inline common/trainers/relevance_transfer_trainer.py:91]: Is this okay? `rnn_outs` from `torch.sigmoid`?

**Review by daemon — APPROVED** [review]

File: common/trainers/relevance_transfer_trainer.py (hunk 1 above)
Comment (achyudh) [reply common/trainers/relevance_transfer_trainer.py:91]: Yeah, this has been a part of the codebase for a while now.
